core/src/sumeragi/fault.rs

```diff
@@ -630,21 +280,1363 @@ pub fn run_sumeragi_main_loop<F>(
+            atomic_sumeragi_guard.latest_block_height = initial_block_height;
+            atomic_sumeragi_guard
+                .current_topology
+                .apply_block(initial_latest_block);
+        } else if !atomic_sumeragi_guard
+            .current_topology
+            .is_consensus_required()
+        {
+            let genesis_network = atomic_sumeragi_guard.genesis_network.take().unwrap();
+            iroha_logger::debug!("Starting commit genesis. Since consensus is not required.");
+
+            iroha_logger::info!("Initializing iroha using the genesis block.");
+
+            atomic_sumeragi_guard.current_topology = sumeragi
+                .try_get_online_topology(&atomic_sumeragi_guard.current_topology)
+                .expect("enough peers to pass genesis");
 
-    loop {
-        println!("Sumeragi is running");
-        std::thread::sleep(Duration::from_secs(1));
+            assert!(!atomic_sumeragi_guard
+                .current_topology
+                .is_consensus_required());
+
+            println!("Genesis being submitted by {}", sumeragi.peer_id.public_key);
+            println!("Online peers are {:?}", sumeragi.get_online_peer_keys());
+
+            atomic_sumeragi_guard.latest_block_height = 0;
+
+            assert_eq!(
+                atomic_sumeragi_guard
+                    .current_topology
+                    .role(&sumeragi.peer_id),
+                Role::Leader
+            );
+            let transactions = genesis_network.transactions.clone();
+            {
+                // Don't start genesis round. Instead just commit the genesis block.
+                if transactions.is_empty() {
+                    panic!("Genesis transactions set contains no valid transactions");
+                } else {
+                    let mut wsv_guard = sumeragi.wsv.lock().unwrap();
+                    let block = PendingBlock::new(transactions, Vec::new())
+                        .chain_first_with_genesis_topology(
+                            atomic_sumeragi_guard.current_topology.clone(),
+                        );
+
+                    {
+                        info!(block_hash = %block.hash(), "Publishing genesis block.");
+
+                        let block = block.validate(&sumeragi.transaction_validator, &wsv_guard);
+
+                        info!(
+                            peer_role = ?atomic_sumeragi_guard.current_topology.role(&sumeragi.peer_id),
+                            block_hash = %block.hash(),
+                            "Created a block",
+                        );
+                        for event in Vec::<Event>::from(&block) {
+                            trace!(?event);
+                            drop(sumeragi.events_sender.send(event));
+                        }
+                        let signed_block = block
+                            .sign(sumeragi.key_pair.clone())
+                            .expect("Sign genesis block.");
+                        {
```

Comment:
We don't *need* all these scopes. But it was constructed this way because the scopes loosely reflect what would previously been function calls. These should be refined over time. But they communicate things about what is connected to what.

Reply:
Ok, thanks you, for clarification!